Player/player.gd

```diff
@@ -414,6 +418,7 @@ func pick_up_weapon(new_weapon) -> void:
     weapon.rotation = 0
```

Comment:
If you want to, you could move the code here that modifies weapon variables into a `weapon.pick_up` method so the player doesn't have to know that much about weapon variables.